packages/medusa-plugin-auth/src/auth-strategies/firebase/admin.ts

```diff
@@ -0,0 +1,48 @@
+import passport from 'passport';
+import { ExtractJwt, Strategy as FirebaseStrategy } from 'passport-firebase-jwt';
+import { ConfigModule, MedusaContainer } from '@medusajs/medusa/dist/types/global';
+import { Router } from 'express';
+import { FIREBASE_ADMIN_STRATEGY_NAME, FirebaseAuthOptions, Profile } from './types';
+import { PassportStrategy } from '../../core/passport/Strategy';
+import { validateAdminCallback } from '../../core/validate-callback';
+import { firebaseAuthRoutesBuilder } from './utils';
+import { auth } from 'firebase-admin';
+
+export class FirebaseAdminStrategy extends PassportStrategy(FirebaseStrategy, FIREBASE_ADMIN_STRATEGY_NAME) {
+	constructor(
+		protected readonly container: MedusaContainer,
+		protected readonly configModule: ConfigModule,
+		protected readonly strategyOptions: FirebaseAuthOptions
+	) {
+		super({
+			jwtFromRequest: strategyOptions.store.jwtFromRequest ?? ExtractJwt.fromAuthHeaderAsBearerToken()
+		});
+	}
+
+	async validate(token: string): Promise<null | { id: string }> {
+		if(this.strategyOptions.admin.verifyCallback) {
+			return await this.strategyOptions.admin.verifyCallback(this.container, token);
+		}
+
+		const decodedToken = await auth().verifyIdToken(token);
```

Comment:
todo: I suggest to decode the token first in the method and then pass it to the verifyCallback. It means also updating the optional verifyCallback typings to pass the entire object. That way, the user does not have to decode it himself. wdyt?

Reply:
I actually wanted to do this today as well.